src/idb_companion.js

```diff
@@ -0,0 +1,42 @@
+var childProcess = require('child_process');
+var process = require('process');
+
+export default {
+    boot (udid, timeout) {
+        var { rc, stdout } = this._exec(['--boot', udid], { timeout });
+
+        if (rc !== 0) {
+            console.error('Failed to boot simulator');
+            console.error(stdout.toString());
+            process.exit(rc);
+        }
+    },
+    shutdown (udid, timeout) {
+        this._exec(['--shutdown', udid], { timeout });
+    },
+    list () {
+        var { rc, stdout } = this._exec(['--list 1']);
+
+        if (rc === 0)
+            return stdout.toString().trim().split('\n');
+        return [];
+    },
+    _exec (args, opts = {}) {
+        try {
+            var execOpts = { stdio: ['pipe', 'pipe', 'ignore' ] };
+
+            if ('timeout' in opts)
+                execOpts['timeout'] = opts.timeout;
+            var stdout = childProcess.execSync(`idb_companion ${args.join(' ')}`, execOpts);
```

Comment:
This library can be used for making arguments parseable in a shell command: https://www.npmjs.com/package/shell-quote

Probably not necessary though, because we control what goes into the execs (ie: users don't provide that).

Reply:
I was thinking the same; args are not user provided so I was ok not sanitizing.